### ISSUE TITLE
"Error looking up service account" issue when enabling the Sentinel

### ISSUE THREAD

**Issue by chdeliens** [bitnami/charts#22541]
### Name and Version

bitnami/redis 18.7.1

### What architecture are you using?

None

### What steps will reproduce the bug?

1. Run `helm repo add bitnami https://charts.bitnami.com/bitnami`;
2. Run `helm install redis bitnami/redis --set sentinel.enabled=true` to deploy Redis with the Sentinel enabled;
3. Run `kubectl get events`
4. Error `create Pod redis-node-0 in StatefulSet redis-node failed error: pods "redis-node-0" is forbidden: error looking up service account default/redis: serviceaccount "redis" not found`

### Are you using any custom parameters or values?

`--set sentinel.enabled=true`

### What is the expected behavior?

Redis is started via two StatefulSets (`redis-master` and `redis-replica`) and use the two (automatically) created service accounts `redis-master` and `redis-replica`.

### What do you see instead?

Redis is started via one StatefulSet (`redis-node`), two service accounts (`redis-master` and `redis-replica`) are created and the `redis-node` StatefulSet wants to use `redis` as its ServiceAccount which fails (since it doesn't exist).

### Additional information

It looks like when the split of service accounts occurred back in Sep 2022, the Sentinel "version" didn't get updated.

IMO, the best way to fix this issue, as suggested in the expected behaviour, would be to have two StatefulSets instead of one.

I'm currently working around the issue with: `helm install redis bitnami/redis --set sentinel.enabled=true --set master.serviceAccount.name=redis --set replica.serviceAccount.name=redis --set replica.serviceAccount.create=false`

**Comment by BenHesketh21** [comment]
Sorry if I misunderstand but if you go to 2 statefulsets instead of one, don't you lose the ability to failover master nodes using sentinel in the event the master goes down?

**Comment by chdeliens** [comment]
> Sorry if I misunderstand but if you go to 2 statefulsets instead of one, don't you lose the ability to failover master nodes using sentinel in the event the master goes down?

Haven't tested if a Sentinel could promote a replica to master in the event you describe. My suggestion of having two statefulset (as when the Sentinel is disabled) was purely to make the Chart maintenance easier.

Now, if that's not working, we'd have to go back to managing a single Service Account instead of two and reference that one instead.

**Comment by BenHesketh21** [comment]
Yeah we use sentinel failovers that quite a lot, it makes Redis much more reliable during cluster upgrades etc.

**Comment by anthosz** [comment]
Just spotted the same issue, seems related to the last release

**Comment by carrodher** [comment]
Thank you for opening this issue and submitting the associated Pull Request. Our team will review and provide feedback. Once the PR is merged, the issue will automatically close.

Your contribution is greatly appreciated!

**Comment by anthosz** [comment]
Dear @carrodher ,

Do you have an ETA? This issue seems more/less blocker at this time :)

**Comment by carrodher** [comment]
Hi, unfortunately, no ETA at this moment. The addition of a new container image into a Helm chart is something that need some internal changes as well in our automation to take that new image into account to update it when there is a new version.
Because of that, the required work on our side could take some time depending on other initiatives the team is working on.

**Comment by chdeliens** [comment]
> Dear @carrodher ,
> 
> Do you have an ETA? This issue seems more/less blocker at this time :)

I'm currently working around the issue with: `helm install redis bitnami/redis --set sentinel.enabled=true --set master.serviceAccount.name=redis --set replica.serviceAccount.name=redis --set replica.serviceAccount.create=false`

This may help you in the meantime :)

**Comment by anthosz** [comment]
Any news on a fix/rollback for this issue?

**Comment by anthosz** [comment]
Tested with the latest release: all ok

Seems this ticket can be closed

**Comment by carrodher** [comment]
This issue has been inactive for a long time and is being closed as part of routine maintenance. Due to other higher-priority work, we weren’t able to investigate this further. If the issue is still relevant and you’re interested in contributing a fix, we’d be happy to review a pull request.

Thanks for your understanding.
